src/Main.elm

```diff
@@ -101,8 +97,36 @@ init location =
 type Msg
     = SelectMovie
     | MovieSelected ( Maybe Movie, List Movie )
+    | FocusMovie MovieSelection
```

Comment:
Preface: I _thought_ I had commented on this a couple of times before, but now I can't find any record of those comments, and they never came up, so I'm wondering if I forgot to submit them or I just don't understand GitHub's PR UI. Anyway, if I'm repeating myself here, I apologize. 

I think the code would be clearer & more maintainable if we split `FocusMovie MovieSelection` into two messages: ShowMovie Movie` and `CloseModal`. A big part of making maintainable elm is using types to narrow the set of possible values that can be used in a given context to something close to the minimum necessary. This reduces ambiguity in the flow of data through the app. 

The way this message is structured now, `FocusMovie` basically acts as a setter for the `focusedMovie` property of the model. That means that if `focusedMovie` ever got in a bad state, to figure out how it got there, a dev would have to explore all views and commands for use of `FocusMovie`, because any view or command could result in that message and change the state. That's workable for the current app size, but as it grows bigger (maybe it won't) that could get out of hand.

The approach I'm suggesting does mean that `Msg` has more values, but I think it's better to have an explicit enumeration of the ways things can change in `Msg` than to have that information implicitly spread out in the view functions where people have to search for them. After all, if we didn't want to have to create new `Msg` values, we could just define:

```elm
type Msg
    = UpdateModel Model
```
and just let the views update model state and pass it back to update, which is now a one-liner except where side effects are involved.

This is the only thing I see that I think is a notable issue with the code as is. It's probably tractable for the app in it's current form, but I wanted to point out that I think it can become a problem in larger apps. 

On the other hand `ShowMovie Movie` can only ever transport a movie. That means the only thing it can do to `focusedMovie` is set it to the `Selected` state. So if we're debugging an issue that doesn't involve `focusedMovie` being set to `Selected`, then we can ignore that message for the purpose of tracking down the issue.

Reply:
Makes sense I think. Tried again.